ARCHITECTURE.md

```diff
@@ -1 +1,83 @@
+
+As you can see, this is a really simplistic overview of how things work. Those boxes inside the blue container belongs
+to the client-side and, inside the orange container, are the server-side ones. Now, boxes with a dotted border are the
+vendors that we interface with, accessed through each platform's APIs (Flutter SDK and Firebase SDK, in this case), and
+colored boxes live here in this repository, they are our beloved source files.
+
+### `/flutter`
+
+Inside the `/flutter` container, we have all of our `Dart` files, which composes almost everything that we need to
+create a Flutter hybrid application. There are some `Custom platform-specific code` which can be eventually required,
+called through platform channels in Dart. Everything else is handled by the `Flutter Framework`, which is used/uses all
+supported platforms, like `iOS` and `Android`.
+
+### `/firebase`
+
+`JavaScript` is to `/firebase`, just like `Dart` is to `/flutter`, even though you can see that we use only `TypeScript`
+- that's because `TypeScript` is just a superset on top of the `JavaScript`, at the end, it outputs `JavaScript` files.
+But, here's the catch: because the `Firebase/GCP Infrastructure` provides so much built-in functionality, there is a
+bunch of stuff that we can do without writing a single line of code. To configure behaviors of these complex services
+provided to us through a simple API, we use some `Configuration Files` - things like rules of our database
+(`firestore.rules` in our case).
+
+### One last thing
+
+... but not less important: this is **heavily influenced** by a bunch personal opinions and experiences in some
+production projects that the team has worked on. This project's external dependencies will keep changing as the time
+goes on, Flutter and Firebase will also keep evolving, and we have to adapt in a way to maintain consistency, integrity
+and scalability of our solution. So, it's probable that there is (and will be) better ways to achieve the same
+goals/objectives. For this, we look into your help to make this project's architecture continuously provide a good
+developer experience to add new features, update old ones and keep those nasty bugs away.
+
+## `.vscode/`
+
+While this project heavily enforces that vscode should be used, IntelliJ is also an alternative, although it won't
+provide the best experience with the setup made in this repository. If you still prefer to use it, there should be no
+problem at all, just make sure to follow the same guidelines specified in
+[Flutter's `settings.json`](flutter/.vscode/settings.json) and
+[Firebase's `settings.json`](firebase/.vscode/settings.json), both which are prepared to automatically apply the
+standard style + format to all Memo's source files.
+
+All configuration files lives under the respective project's `.vscode` folder and - while there are some that will
+disagree with this decision - **should be git-tracked**.
```

Comment:
NIT:

```suggestion
All configuration files lives under the respective project's `.vscode`.
```

Reply:
Ditto:
I think this goes the way of https://github.com/olmps/memo/pull/194#discussion_r717556085

If you disagree, we can discuss about it, I'm open to it.

Reply:
About this specific one, I think it could be rewritten to emphasis why we are the team the check it in source control, instead pointing out that some may disagree. Suggestion:

```suggestion
All configuration files lives under the respective project's `.vscode` folder and **should be git-tracked**. There are open discussions about the git-tracking of `.vscode` folder, and we are those that aim for standardize our codebase by using lint and defining IDE standards.
```

Or something like that

Reply:
About this specific one, I think it could be rewritten to emphasis why we are the team the check it in source control, instead pointing out that some may disagree. Suggestion:

```suggestion
All configuration files lives under the respective project's `.vscode` folder and **should be git-tracked**. There are open discussions about the git-tracking of `.vscode` folder, and we are those that aim for standardize our codebase by using lint and defining IDE standards.
```

Or something like that

Reply:
Removed "while there are some that will disagree with this decision" to keep this compact but remove the justification as to why